File: stars-app/frontend/src/types/cahps.ts
export interface CahpsComposite {
  composite_code: string;
  composite_name: string;
  current_pct: number;
  status: string;
  weight: string;
  part: string;
}

export interface CahpsOverview {
  contract_id: string;
  current_rating: number;
  projected_rating: number;
  gap_to_4_star: number;
  days_remaining: number;
  qbp_at_stake: number;
  composites: CahpsComposite[];
}

File: stars-app/backend/models/cahps.py
from pydantic import BaseModel


class CahpsComposite(BaseModel):
    composite_code: str
    composite_name: str
    current_pct: float
    status: str
    weight: str = "2x"
    part: str = "C"


class CahpsOverviewResponse(BaseModel):
    contract_id: str
    current_rating: float
    projected_rating: float
    gap_to_4_star: float
    days_remaining: int
    qbp_at_stake: float
    composites: list[CahpsComposite]
